src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/events/ManageExternalAppointmentsService.kt

```diff
@@ -108,15 +110,21 @@ class ManageExternalAppointmentsService(
     )
   }
 
-  private fun Collection<AppointmentSearchResult>.findMatching(appointment: PrisonAppointment) =
-    singleOrNull {
+  private fun Collection<AppointmentSearchResult>.findMatchingActivitiesAndAppointments(appointment: PrisonAppointment): List<AppointmentSearchResult> =
```

Comment:
There could potentially be multiple matches e.g. pre, main and post `singleOrNull` would reject this (at least that is what I think is wrong).

Reply:
Sounds a likely candidate.